### PR TITLE
Dc coverage to publication tag

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 1]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                );
+    }
+
+    private static Stream<Arguments> provideMetadataForDate() {
+        String year = "2021";
+        String month = "02";
+        String day = "19";
+        CreatePublicationRequest request = new CreatePublicationRequest();
+        request.setEntityDescription(new EntityDescription.Builder()
+                .withDate(new PublicationDate.Builder()
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:135]: variable

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 2]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
-        startMock(filename);
+    @ParameterizedTest
+    @MethodSource({
+            "provideMetadataForTags",
+            "provideMetadataForAbstract",
+            "provideMetadataForTitle",
+            "provideMetadataForDate",
+            "provideMetadataForContributors"
+    })
+    public void test(String html, CreatePublicationRequest expectedRequest) throws IOException {
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:86]: test naming

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 3]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                .build());
+
+        CreatePublicationRequest yearOnlyrequest = new CreatePublicationRequest();
+        yearOnlyrequest.setEntityDescription(new EntityDescription.Builder()
+                .withDate(new PublicationDate.Builder()
+                        .withYear(year)
+                        .build())
+                .build());
+
+        String date = String.join("-", year, month, day);
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:149]: Extract as named method, it is an annoyance — use now()

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 4]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                        "dc.creator", name),
+                        request),
+                arguments(Map.of(
+                        "DC.creator", name),
+                        request)
+                );
+    }
+
+    private static Stream<Arguments> provideMetadataForDate() {
+        String year = "2021";
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:130]: Extract as method, use now()

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 5]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                arguments(Map.of(
+                        "DC.date", year),
+                        yearOnlyrequest)
+                );
+    }
+
+    private static Stream<Arguments> provideMetadataForTitle() {
+        String title = "Title";
+
+        CreatePublicationRequest request = new CreatePublicationRequest();
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:170]: Extract method

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 6]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                        "DC.title", title),
+                        request)
+                );
+    }
+
+    private static Stream<Arguments> provideMetadataForTags() {
+        String coverage = "Coverage";
+        String subject = "Subject";
+
+        CreatePublicationRequest request = new CreatePublicationRequest();
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:189]: Extract method

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 7]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                        "dc.subject", subject),
+                        request)
+                );
+    }
+
+    private static Stream<Arguments> provideMetadataForAbstract() {
+        String description = "Description";
+        String abstractString = "Abstract";
+
+        CreatePublicationRequest request = new CreatePublicationRequest();
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:206]: Extract methods

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 8]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+
+        assertThat(actual, is(equalTo(expectedRequest)));
+
+        wireMockServer.stop();
+    }
+
+    private static Stream<Arguments> provideMetadataForContributors() throws MalformedContributorException {
+        String name = "Full name";
+
+        CreatePublicationRequest request = new CreatePublicationRequest();
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:103]: Extract method

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 4 above)
Comment (janbj) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:145]: Why is it better to use now()?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 9]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+            "provideMetadataForDate",
+            "provideMetadataForContributors"
+    })
+    public void test(String html, CreatePublicationRequest expectedRequest) throws IOException {
+        String filename = "article.html";
+        URI uri = prepareWebServerAndReturnUriToMetadata(filename, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+
+        CreatePublicationRequest actual = request.get();
```
[the remark this answers is not in the document]
Comment (janbj) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:100]: What warning?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 10]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+                    .withIdentity(new Identity.Builder()
+                            .withName(name)
+                            .build())
+                    .build())
+                )
+                .build());
+
+        return Stream.of(
+                arguments(Map.of(
+                        "dc.contributor", name),
```
[the remark this answers is not in the document]
Comment (brinxmat) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:114]: Still prefer constants

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 11]
```diff
@@ -64,16 +75,196 @@ public void getMetadataJsonFromUpperCaseTagMetadataReturnsCreatePublicationReque
...
+        metadata.keySet().stream().forEach(property -> {
+            builder.append(String.format(tagTemplate, property, metadata.get(property)));
+        });
+        builder.append(String.format(tagTemplate, "test", "test"));
+        builder.append(bottom);
+        return builder.toString();
+    }
+
+    private URI prepareWebServerAndReturnUriToMetadata(String filename) {
+        var body = getBody("/" + filename);
```
[the remark this answers is not in the document]
Comment (brinxmat) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:286]: I still expect constants :)

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**